### ISSUE TITLE
Create high-level API to append data to a file

### ISSUE THREAD

**Issue by hgupta9** [hgupta9/FluentFTP#96]
Currently one must download a file, append to it and then use `UploadFile(..FtpExists.Append)`. This is stupid. So we need:

- AppendFile(string remotePath, string data)
- AppendFile(string remotePath, Stream data)
- AppendFile(string remotePath, byte[] data)

**Comment by jblacker** [comment]
Not that I disagree that this would be helpful, but I wouldn't say that a file really needs to be downloaded to be appended when `OpenAppend()` is exposed and available.  (unrelated, but I should have a PR for you this weekend)

**Comment by hgupta9** [comment]
Exactly. Currently the only API to append to a file is OpenAppend() which needs GetReply() and stream.Close() and stuff. There should be a high-level API which simply takes the given data and appends it to the file on the server without needing to download the file first. This can be accomplished by checking the length of the file and then appending the given data after that. Not very hard.

**Comment by superoctave2** [comment]
+1 to this enhancement idea.  Upload() with FtpExists.Append, because of the built-in seek of the source file discussed in https://github.com/robinrodricks/FluentFTP/issues/213, means there's no high-level method facilitating an "append multiple source files to the target" use case (Upload() appears to have only been designed for a "periodically sync the target file with a continually-growing source file like a log" use case.)  One can use OpenAppend() to do it as you say, but it'd be great to have some of that messiness abstracted away.

**Comment by robinrodricks** [comment]
Implemented by UploadFile.
